### PR TITLE
chore: log shutdown reason for check_oom trace log

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes <issue-or-jira-number>

```
2024-06-27T03:03:08.764259+00:00 [debug] clientid: test, msg: check_oom, peername: 172.100.239.1:42259, username: emqx_u1, policy: #{max_heap_size => 4194304,enable => true,max_mailbox_size => 1000}
```
The log is missing essential information, and 'active_n (100)' is alternately printed in the trace log. Adding 'incoming_pubs', 'incoming_bytes', and result meta fields to enhance readability.

```
2024-06-27T03:49:49.578308+00:00 [debug] clientid: test, msg: check_oom_ok, mfa: emqx_connection:check_oom/3(1104), peername: 172.100.239.1:42293, username: emqx_u1, result: ok, incoming_bytes: 33, incoming_pubs: 1, policy: #{max_heap_size => 4194304,enable => true,max_mailbox_size => 1000}
```
Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_gateway_ocpp/src/emqx_ocpp_connection.erl [hunk 1]
```diff
@@ -469,20 +469,18 @@ websocket_handle({Frame, _}, State) ->
...
+    return(ensure_rate_limit(NState));
 websocket_info({cast, Msg}, State) ->
     handle_info(Msg, State);
 websocket_info({incoming, Packet}, State) ->
     handle_incoming(Packet, State);
 websocket_info({outgoing, Packets}, State) ->
     return(enqueue(Packets, State));
-websocket_info({check_gc, Stats}, State) ->
-    return(check_oom(run_gc(Stats, State)));
+websocket_info({check_gc, Cnt, Oct}, State) ->
```
Comment (qzhuyan) [inline apps/emqx_gateway_ocpp/src/emqx_ocpp_connection.erl:482]: while support hot upgrade, maybe we need to keep the old clause `websocket_info({check_gc, States}, State)` to handle old events from the queue or purge.

**Review by qzhuyan — APPROVED** [review]

File: apps/emqx/src/emqx_connection.erl [hunk 2]
```diff
@@ -1080,25 +1077,36 @@ retry_limiter(#state{channel = Channel, limiter = Limiter} = State) ->
...
+            ?tp(warning, check_oom_shutdown, #{
+                policy => ShutdownPolicy,
+                incoming_pubs => Pubs,
+                incoming_bytes => Bytes,
+                shutdown => Reason
+            }),
             erlang:exit({shutdown, Reason});
-        _ ->
+        Result ->
+            ?tp(debug, check_oom_ok, #{
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_connection.erl:1104]: 🤔 maybe we will log too much here. Prefer to have this enable/disabled by a compiler flags.

File: apps/emqx_gateway_ocpp/src/emqx_ocpp_connection.erl (hunk 1 above)
Comment (zhongwencool) [reply apps/emqx_gateway_ocpp/src/emqx_ocpp_connection.erl:482]: we don't support hot upgrade yet.

File: apps/emqx/src/emqx_connection.erl (hunk 2 above)
Comment (zhongwencool) [reply apps/emqx/src/emqx_connection.erl:1104]: active_n(100) packet log once. I think this is ok.

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3406)

**Pull request merged**